crates/linker/src/pe.rs

```diff
@@ -392,14 +392,14 @@ pub(crate) fn surgery_pe(executable_path: &Path, metadata_path: &Path, roc_app_b
                 } else if name == "___chkstk_ms" {
                     // this is a stack probe that is inserted when a function uses more than 2
                     // pages of stack space. The source of this function is not linked in, so we
-                    // have to get a bit creative: we just jump to a `ret` instruction, so this
-                    // function call becomes a no-op.
+                    // have to do it ourselves. We patch in the bytes as a separate section, and
+                    // here just need to jump to those bytes
 
-                    // the last byte of the section should be a `ret` instruction
-                    assert_eq!(slice.last(), Some(0xc3).as_ref());
+                    // This relies on the ___CHKSTK_MS section being the last text section in the list of sections
+                    let destination = length - ___CHKSTK_MS.len();
```

Comment:
Well the linker is only making an assumption about its own behaviour so it's OK.
The only downside I can think of with this approach is that it might not scale neatly if we ever had to do more things like this. We'd have a convention for the last section, second last, etc.
But until we run into that, I think it's a good solution!